app/controllers/thredded/autocomplete_users_controller.rb

```diff
@@ -6,17 +6,27 @@ class AutocompleteUsersController < Thredded::ApplicationController
     def index
       authorize_creating PrivateTopicForm.new(user: thredded_current_user).private_topic
       users = params.key?(:q) ? users_by_prefix : users_by_ids
+
       render json: {
-        results: users.map do |user|
-          { id:         user.id,
-            name:       user.send(Thredded.user_name_column),
-            avatar_url: Thredded.avatar_url.call(user) }
-        end
+          results: users.map do |user|
+            { id:         user.id,
+              name:       display_name(user),
+              avatar_url: Thredded.avatar_url.call(user) }
+          end
       }
     end
 
     private
 
+    # Defines how the name is displayed in the autocomplete search results
+    def display_name(user)
+      if Thredded.user_display_name_method == :to_s
+        user.send(Thredded.user_name_column)
+      else
+        "#{user.send(Thredded.user_name_column)} - #{user.send(Thredded.user_display_name_method)}"
+      end
+    end
+
```

Comment:
This is not the display name, this is only meant to be used for displaying the autocomplete results in @-mentions and private message recipients, so I don't think `Thredded.display_name` would be the right name for the proc.

Also, maybe we don't need this to be a setting. It's probably rare to want to customize it, and one can always override the controller method if they need to.